### PR TITLE
[JENKINS-70240] fix for non http based URLs in PluginManager.doCheckUpdateSiteUrl

### PR DESCRIPTION
**Pull request by jtnord**
update sites are not restricted to just http/https file based ones are also common.  
Add support for file based url schemes in the doCheck to avoid an angry jenkins.

code will return a FormValidation error in the case an unsupported scheme is now used rather than an angry jenkins.
Update the help and translations.

subsumes #7520 

See [JENKINS-70240](https://issues.jenkins.io/browse/JENKINS-70240).

TODO: fix up stupid code charset mangling, and Manual dev testing still to be performed. (by the author)

### Proposed changelog entries

- Prevent Angry Jenkins when checking a non http(s) based update center URL.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7524"></a>

### CODE REVIEW

File: core/src/main/resources/hudson/Messages.properties
Comment (jtnord) [inline core/src/main/resources/hudson/Messages.properties:25]: this is fun...   all needs to be reverted...

File: core/src/main/resources/hudson/Messages_zh_TW.properties
Comment (jtnord) [inline core/src/main/resources/hudson/Messages_zh_TW.properties:79]: :huh:

**Review by basil — APPROVED** [review]
Code looks good to me; I fixed up the localizations and verified the new automated test passes.

**Review by rsandell — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
